gradle-plugin/src/main/java/com/google/cloud/tools/dependencies/gradle/LinkageCheckTask.java

```diff
@@ -217,16 +217,27 @@ private void recordDependencyPaths(
         ResolvedDependencyResult resolvedDependencyResult =
             (ResolvedDependencyResult) dependencyResult;
         ResolvedComponentResult child = resolvedDependencyResult.getSelected();
-        stack.add(child);
-        recordDependencyPaths(output, stack, targetCoordinates);
+
+        if (stack.contains(child)) {
+          // Circular dependency check
+          getLogger()
+              .error(
+                  "Circular dependency for: "
+                      + resolvedDependencyResult
+                      + "\n The stack is: "
+                      + ImmutableList.copyOf(stack));
```

Comment:
Probably don't need the defensive copy here.

Reply:
That's right. Removed.